design-proposals/flavors-and-preferences.md

```diff
@@ -0,0 +1,475 @@
+      * [Cdrom](http://kubevirt.io/api-reference/main/definitions.html#_v1_cdromtarget)
+        * Bus
+          * `VirtualMachinePreferencesSpec.Devices.PreferredCdRomBus`
+        * readonly
+          * None
+        * Tray
+          * None
+      * dedicatedIOThread
+        * `VirtualMachineFlavorSpec.Devices.dedicatedIOThread`
+      * [Disk](http://kubevirt.io/api-reference/main/definitions.html#_v1_disktarget)
+        * Bus
+          * `VirtualMachinePreferencesSpec.Devices.PreferredDiskBus`
+        * pciAddress
+          * None
+        * readonly
+          * None
+      * io
+        * `VirtualMachineFlavorSpec.Devices.PreferredIOMode`
+      * [Lun](http://kubevirt.io/api-reference/main/definitions.html#_v1_luntarget)
+        * bus
+          * `VirtualMachinePreferencesSpec.Devices.PreferredLunBus`
+        * readOnly
+          * None
+      * name
+        * None
+      * serial
+        * None
+      * shareable
+        * None
+      * tag
+        * None
+      * [Filesystems](http://kubevirt.io/api-reference/main/definitions.html#_v1_filesystem)
+        * Name
+          * None
+        * virtiofs
+          * None
+      * [Gpus](http://kubevirt.io/api-reference/main/definitions.html#_v1_gpu)
+        * deviceName
+          * `VirtualMachineFlavorSpec.Devices.GPUs`
+        * name
+          * None
+        * tag
+          * None
+        * virtualGPUOptions
+          * `VirtualMachinePreferencesSpec.Devices.virtualGPUOptions`
+      * [HostDevices](http://kubevirt.io/api-reference/main/definitions.html#_v1_hostdevice)
+        * deviceName
+          * `VirtualMachineFlavorSpec.Devices.HostDevices`
+        * name
+          * None
+        * tag
+          * None
+      * [Inputs](http://kubevirt.io/api-reference/main/definitions.html#_v1_input)
+        * bus
+          * `VirtualMachinePreferencesSpec.Devices.PreferredInputBus`
+        * Name
+          * None
+        * Type
+          * `VirtualMachinePreferencesSpec.Devices.PreferredInputType`
+      * [Interfaces](http://kubevirt.io/api-reference/main/definitions.html#_v1_interface)
+        * bootOrder
+          * None
+        * bridge
+          * None
+        * dhcpoptions
+          * None
+        * macaddress
+          * None
+        * [macvtap](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfacemacvtap) 
+          * None
+        * [masquerade](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfacemasquerade) 
+          * None
+        * model
+          * `VirtualMachinePreferencesSpec.Devices.Interfaces.PreferredModel`
+        * name
+          * None
+        * pciAddress
+          * None
+        * ports
+          * None
+        * [slirp](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfaceslirp) 
+          * None
+        * [sriov](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfacesriov) 
+          * None
+        * Tag
+          * None
```

Comment:
By default, no tags will be generated, since the original idea was to let users tag only the relevant devices with a tag that would be meaningful to the guest user (i.e "fastNic").
I think generating these tags using the device name, in a consistent way, would work. We should also have a way to not generate the tags at all.